aleph-client/src/staking.rs

```diff
@@ -271,10 +272,21 @@ pub fn bonded<C: AnyConnection>(connection: &C, stash: &KeyPair) -> Option<Accou
         .unwrap_or_else(|_| panic!("Failed to obtain Bonded for account id {}", account_id))
 }
 
-pub fn ledger<C: AnyConnection>(
-    connection: &C,
-    controller: &KeyPair,
-) -> Option<pallet_staking::StakingLedger<AccountId, Balance>> {
+/// Since PR #10982 changed `pallet_staking::StakingLedger` to be generic over
+/// `T: pallet_staking::Config` (somehow breaking consistency with similar structures in other
+/// pallets) we have no easy way of retrieving ledgers from storage. Thus, we chose cloning
+/// (relevant part of) this struct instead of implementing `Config` trait.
```

Comment:
I don't quite get the problem here. What is the problem with implementing the `Config` trait?

Also, I assume this will break as soon as there is a migration changing the shape of the data in any way, right? I don't think this is a major problem, just making sure I understand.

Reply:
The problem is that implementing `Config` trait is quite cumbersome - it has very broad API and this would seem very strange that `aleph-client` provides `pallet_x::Config` only for `x = staking`, which is used simply for reading a secondary struct from storage.

Yes, this will break as soon as there is a change influencing decoding of this structure.

Personally, I would rather make a PR to Substrate with reverting this parametrization instead of adjusting to it #rebel